File: backend/app/schemas/progress.py
from typing import Optional
from pydantic import BaseModel


class WeeklySummaryResponse(BaseModel):
    period_start: str
    period_end: str
    avg_intake_kcal: Optional[float]
    avg_target_kcal: Optional[float]
    total_intake_kcal: Optional[float]
    days_on_target: int
    days_logged: int
    consistency_score: Optional[float]


class MonthlySummaryResponse(BaseModel):
    period_start: str
    period_end: str
    avg_intake_kcal: Optional[float]
    avg_target_kcal: Optional[float]
    total_intake_kcal: Optional[float]
    days_on_target: int
    days_logged: int
    consistency_score: Optional[float]


class StreakResponse(BaseModel):
    current_streak: int
    longest_streak: int


class ConsistencyResponse(BaseModel):
    score: float
    days_on_target: int
    total_days: int
    period: str

File: frontend/src/types/progress.ts
export interface WeeklySummary {
  period_start: string;
  period_end: string;
  avg_intake_kcal: number | null;
  avg_target_kcal: number | null;
  total_intake_kcal: number | null;
  days_on_target: number;
  days_logged: number;
  consistency_score: number | null;
}

export interface StreakData {
  current_streak: number;
  longest_streak: number;
}

export interface ConsistencyData {
  score: number;
  days_on_target: number;
  total_days: number;
  period: string;
}
